src/sources/mongodb_metrics/mod.rs

```diff
@@ -277,6 +280,10 @@ impl MongoDbMetrics {
             .run_command(command, None)
             .await
             .map_err(CollectError::Mongo)?;
+        emit!(&BytesReceived {
+            byte_size: std::mem::size_of_val(&doc),
```

Comment:
I don't think this is right. `std::mem::size_of_val` will give you the size of the type, that is, [bson](https://docs.rs/bson/2.1.0/bson/index.html)::[document](https://docs.rs/bson/2.1.0/bson/document/index.html)::[Document](https://docs.rs/bson/2.1.0/bson/document/struct.Document.html#), rather than the bytes actually fetched from the MongoDB server. I think you'll need to look for a way to intercept the request and count the bytes. I didn't see anything with a cursory look over https://docs.rs/mongodb/latest/mongodb/index.html though. If you can't find anything either, that might mean we'll need to punt on this metric for this source for now and open an issue on the upstream repo.

Reply:
I'll remove it then 😅